src/bus/three_wire.rs

```diff
@@ -0,0 +1,128 @@
+#![allow(clippy::inconsistent_digit_grouping, clippy::unusual_byte_groupings)]
+
+use core::fmt;
+use embedded_hal::blocking::spi::{Transfer, Write};
+
+use crate::bus::Bus;
+
+const WRITE_MODE_MASK: u8 = 0b00000_1_0;
+
+const FIXED_DATA_LENGTH_MODE_1: u8 = 0b000000_01;
+const FIXED_DATA_LENGTH_MODE_2: u8 = 0b000000_10;
+const FIXED_DATA_LENGTH_MODE_4: u8 = 0b000000_11;
+
+// TODO This name is not ideal, should be renamed to FDM
+pub struct ThreeWire<Spi: Transfer<u8> + Write<u8>> {
+    spi: Spi,
+}
+
+impl<Spi: Transfer<u8> + Write<u8>> ThreeWire<Spi> {
+    pub fn new(spi: Spi) -> Self {
+        Self { spi }
+    }
+
+    pub fn release(self) -> Spi {
+        self.spi
+    }
+}
+
+impl<Spi: Transfer<u8> + Write<u8>> Bus for ThreeWire<Spi> {
+    type Error = ThreeWireError<<Spi as Transfer<u8>>::Error, <Spi as Write<u8>>::Error>;
+
+    /// Transfers a frame with an arbitrary data length in FDM
+    ///
+    /// This is done by passing chunks of fixed length 4, 2, or 1.  For example if a frame looks like this:
+    ///
+    /// (address 23) 0xF0 0xAB 0x83 0xB2 0x44 0x2C 0xAA
+    ///
+    /// This will be sent as separate frames in the chunks
+    ///
+    /// (address 23) 0xF0 0xAB 0x83 0xB2
+    /// (address 27) 44 2C
+    /// (address 29) AA
```

Comment:
What is the advantage of this behavior?

Reply:
These two modes are for when you either have or do not have a CS pin.  You can connect with only three wires (MISO, MOSI, CLK), but you have to use FDM.